crates/evm/fuzz/src/strategies/state.rs

```diff
@@ -245,12 +245,26 @@ impl FuzzDictionary {
             // Insert push bytes
             if let Some(code) = &account_info.code {
                 self.insert_address(*address);
-                self.collect_push_bytes(code.bytes_slice());
+                self.collect_push_bytes(code);
             }
         }
     }
 
-    fn collect_push_bytes(&mut self, code: &[u8]) {
+    fn collect_push_bytes(&mut self, code: &Bytecode) {
+        let mut code = code.original_byte_slice();
+        if code.is_empty() {
+            return;
+        }
+
+        // Remove metadata by looking up the last two bytes of original bytecode.
```

Comment:
is this always true?

Reply:
For initcode, no. Sometimes e.g. long strings in a constructor are after the metadata hash.

For runtime code, yes, IF the metadata hash is present. When `bytecode_hash = none` there is still a metadata hash containing the solidity version, so the last two bytes are still the length. When `cbor_metadata = false` then there is no bytecode hash at all and the last two bytes are code

To know for certain if what you have is a metadata hash, without knowing the compilation settings, you'd assume the last two bytes are the length of the metadata hash, then try CBOR decoding that hash. If it decodes successfully, it's a metadata hash, otherwise it's code. 

Here is some old, probably bad, rust code I had that did this using ciborium: https://github.com/ScopeLift/cove-backend/blob/d8c875c5a64bf89a874892bc8ed57751c47b1233/src/bytecode.rs#L53-L97